### PR TITLE
Tests: use ES modules

### LINKED ISSUE #23304: Some unit tests on web fails because of undefined three
**Describe the bug**

Eight "Examples" unit tests fail on web browsers.

**To Reproduce**

1. `$ npm run test-unit`
2. `$ npm run start`
3. Access `http://localhost:8080/test/unit/UnitTests.html` on a web browser and see the errors.

**Expected behavior**

All the tests pass and no errors.

**Screenshots**

image

three.example.unit.js:17763:20 is

```
return new three.BufferAttribute( array, itemSize, normalized );
```

and `three` is undefined.

I speculate the problem is caused by #23255 because import maps doesn't work corrently for building the "examples" unit tests?

Update: Probably related https://github.com/mrdoob/three.js/pull/23293#issuecomment-1018580295

**Platform:**

 - Device: Desktop
 - OS: Windows
 - Browser: Chrome
 - Three.js version: dev

### PR DESCRIPTION
**Pull request by marcofugaro**
Related issue: Fixes #23304

**Description**

While we're at it, let's update the tests to use es modules introduced in Node 16.
No need to use rollup in tests anymore.

[Following the Node.js docs](https://nodejs.org/api/packages.html#conditional-exports), the new package.json looks like this: 

```json
"type": "module",
"main": "./build/three.js",
"module": "./build/three.module.js",
"exports": {
  ".": {
    "import": "./build/three.module.js",
    "require": "./build/three.js"
  },
  "./examples/jsm/*": "./examples/jsm/*.js",
  "./src/*": "./src/*.js"
},
```

**Note** that I had to include the `src` folder in the exports field as well, since the tests are importing from the `src` folder.

I also added the `.js` extension to imports since in esmodules (both web and node), the extension has to be explicit.

You can test this PR by:

1. make sure you have Node 16 or greater
2. `npm i --prefix test`
3. `npm run test-unit`
4. `npm run test-unit-examples`

To test the browser version
1. `npm i --prefix test`
2. `npm start`
3. Open http://localhost:8080/test/unit/UnitTests.html

**Note** that the web tests work only in Chrome/Edge because of import maps. The import maps polyfill doesn't work with Qunit because it's syncronous.

/cc @takahirox

### CODE REVIEW

**Comment by marcofugaro** [comment]
> Do we still need the files section?

Yes 🙂 that field contains the files that are downloaded when a user installs three from npm.
The `exports` field indicates the files directly available for import.

File: package.json [hunk 1]
```diff
@@ -2,8 +2,17 @@
...
-  "module": "build/three.module.js",
+  "type": "module",
+  "main": "./build/three.js",
+  "module": "./build/three.module.js",
+  "exports": {
+    ".": {
+      "import": "./build/three.module.js",
+      "require": "./build/three.js"
+    },
+    "./examples/jsm/*": "./examples/jsm/*.js",
```
Comment (mrdoob) [inline package.json:13]: What would happen if we renamed this? Say...

```json
"./addons/*": "./examples/jsm/*.js",
```

Would people be able to this?

```js
import { OrbitControls } from 'three/addons/controls/OrbitControls.js'
```

And, would the old version stil work?

```js
import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls.js'
```

Comment (marcofugaro) [reply package.json:13]: Apparently yes, you can do that, from the docs:


> And, would the old version stil work?

Unfortunately, no, node (or any bundler) throws an error if you're trying to import something that is not in the `export` fields.

But you can get around this limitation by having both of the import aliases there:

```json
"./addons/*": "./examples/jsm/*.js",
"./examples/jsm/*": "./examples/jsm/*.js",
```

Comment (mrdoob) [reply package.json:13]: Excellent! That makes thing easy.

What do you think about the name `addons`? Do you think @drcmda will like it? 🤓

Comment (marcofugaro) [reply package.json:13]: Hahaha I don't mind the name!

I even made a [`three-addons`](https://github.com/marcofugaro/three-addons) package back when `examples/jsm` wasn't a thing 😆

Comment (mrdoob) [reply package.json:13]: Okay, I'll keep it in mind 🙏

**Comment by mrdoob** [comment]
Thanks!

File: package.json (hunk 1 above)
Comment (drcmda) [reply package.json:13]: @mrdoob i do like it 😁

**Pull request merged**